File: leetcode/1598-crawler-log-folder/minOperations.go
package leetcode

func minOperations(logs []string) int {
	var step = 0
	for _, log := range logs {
		if log == "./" {
			continue
		}

		if log == "../" {
			if step > 0 {
				step--
			}
			continue
		}

		step++
	}

	return step
}

File: leetcode/1598-crawler-log-folder/minOperations.ts
function minOperations(logs: string[]): number {
  let step = 0;

  for (const log of logs) {
    if (log == "./") {
      continue;
    }
    if (log == "../") {
      if (step > 0) {
        step--;
      }
      continue;
    }

    step++;
  }

  return step;
}

export default minOperations;
